quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/hostname/DefaultHostnameProvider.java

```diff
@@ -314,6 +314,15 @@ public void init(Config.Scope config) {
             throw new RuntimeException("You can not set both '" + HostnameOptions.HOSTNAME_ADMIN.getKey() + "' and '" + HostnameOptions.HOSTNAME_ADMIN_URL.getKey() + "' options");
         }
 
+        final boolean adminShouldBeSameAsFrontend = hostnameEnabled && adminHostName == null && adminBaseUri == null;
+        if (adminShouldBeSameAsFrontend) {
+            if (frontEndBaseUri != null) {
+                adminBaseUri = frontEndBaseUri;
+            } else if (frontEndHostName != null) {
+                adminHostName = frontEndHostName;
+            }
+        }
```

Comment:
This is a good point that currently we actually don't consider any frontend URLs at all when admin is not specified. Initially, I thought it's just about scheme but now I realized it's everything incl. hostname. The original issues says "Admin URL doesn't fully honor `--hostname-url`". Well, it doesn't honor it at all!

I think this change is too big of a breaking change since we'd need to fix everything, not just hostname. It'd add additional mess to already messy behaviour with our current hostname impl, breaking it again for the few people who managed to configure it.

I'd fix this in the new impl we're preparing.

WDYT @mabartos @stianst?

Reply:
> I think this change is too big of a breaking change since we'd need to fix everything, not just scheme. 

Yes, that was the main reason why I was defensive to provide these changes to the 22 release. It'd change the functionality for the present deployments with the usage of Keycloak 23, indeed. AFAIK, this provided behavior is supposed to be the intended default behavior for the admin hostname. 

However, as we're already working on the new hostname implementation, it'd be good to include these changes there and prevent the additional change in the present behavior; as you mentioned. 

So, I'll close this PR and we'll consider it afterward. @vmuzikar WDYT?

Reply:
@mabartos Works for me but I'd like confirmation from @stianst as he reported that issue originally.